### PR TITLE
API helper to create/update a collection add-on

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/4567

### CODE REVIEW

File: src/amo/api/collections.js [hunk 1]
```diff
@@ -285,3 +287,79 @@ export const createCollection = ({
...
+|};
+
+type ModifyCollectionAddonParams =
+  | {| action: 'create', ...CreateCollectionAddonParams |}
+  | {| action: 'update', ...UpdateCollectionAddonParams |};
+
+export const modifyCollectionAddon = (
+  params: ModifyCollectionAddonParams,
+): Promise<void> => {
+  const { action, api, user, collectionSlug, addonId } = params;
```
Comment (bobsilverberg) [inline src/amo/api/collections.js]: Should these be in alphabetical order, or do we not care about that in this case?

File: src/amo/api/collections.js [hunk 2]
```diff
@@ -285,3 +287,79 @@ export const createCollection = ({
...
+    endpoint = `${endpoint}/${addonId}`;
+  }
+
+  return callApi({ auth: true, body, endpoint, method, state: api });
+};
+
+export const createCollectionAddon = ({
+  api,
+  collectionSlug,
+  addonId,
```
Comment (bobsilverberg) [inline src/amo/api/collections.js]: These are out of order too, as are the ones below.

File: src/amo/api/collections.js [hunk 3]
```diff
@@ -285,3 +287,79 @@ export const createCollection = ({
...
+  invariant(collectionSlug, 'The collectionSlug parameter is required');
+  invariant(user, 'The user parameter is required');
+
+  let method = 'POST';
+  const body = { addon: addonId, notes: params.notes || null };
+  let endpoint =
+    `accounts/account/${user}/collections/${collectionSlug}/addons`;
+
+  if (action === 'update') {
+    if (typeof params.notes === 'undefined') {
```
Comment (bobsilverberg) [inline src/amo/api/collections.js]: Why not just `params.notes === 'undefined'`?

Comment (bobsilverberg) [reply src/amo/api/collections.js]: Oops, I of course meant `param.notes === undefined`.

Comment (kumar303) [reply src/amo/api/collections.js]: Oh, heh, it is an old habit from IE6 days where referencing an undefined name would throw. I can change it.

File: src/amo/api/collections.js (hunk 1 above)
Comment (kumar303) [reply src/amo/api/collections.js]: Personally I don't really care (I use an editor with good name finding capabilities) but I can change it. I don't feel strongly about it. I'll adjust the others too.

**Comment by kumar303** [comment]
Thanks for the review. I was a bit surprised that I got [disjoint unions](https://flow.org/en/docs/types/unions/#toc-disjoint-unions) to work. I'm not sure what is different here from your approach on `modifyCollection`. It's a simpler function. Maybe the other one was triggering a subtle bug in Flow.

**Comment by kumar303** [comment]
I know that including `{ action: 'created' | 'updated' }` as part of each union'd object is key but I did try making that change to your `modifyCollection` patch. It didn't seem to help.

**Pull request merged**